haystack/core/pipeline/async_pipeline.py

```diff
@@ -17,16 +17,81 @@
 
 class AsyncPipeline(PipelineBase):
     """
-    Asynchronous version of the orchestration engine.
+    Asynchronous version of the Pipeline orchestration engine.
+
+    Manages the execution of components in a pipeline allowing for concurrent execution
```

Comment:
`Manages components in a pipeline allowing for concurrent processing when the pipeline's execution graph permits.`

Maybe something like those for the first sentence? To avoid having the word `execution` 3 times in the same sentence